### PR TITLE
Fix multiple issues with TextureAtlasVisualiser

### PR DESCRIPTION
**Pull request by peppy**
- Fixes references being held indefinitely (even when not open)
- Fixed bind count being incorrect (was counting usages, not binds)
- Shows bind count as a number
- Changes colour range to show even low usages
- Removes unnecessary `TextureRemoved` logic (relies on `WeakReference` now)
- Renames `TextureAdded` to `TextureCreated`

### CODE REVIEW

File: osu.Framework/Graphics/OpenGL/GLWrapper.cs [hunk 1]
```diff
@@ -359,33 +359,36 @@ internal static void SetActiveBatch(IVertexBatch batch)
...
             last_bound_texture_is_atlas[GetTextureUnitId(unit)] = texture is TextureGLAtlas;
+
+            return didBind;
         }
 
         /// <summary>
         /// Binds a texture to draw with.
         /// </summary>
         /// <param name="textureId">The texture to bind.</param>
         /// <param name="unit">The texture unit to bind it to.</param>
```
Comment (smoogipoo) [inline osu.Framework/Graphics/OpenGL/GLWrapper.cs]: Needs an xmldoc for the return value. Something along the lines of:

`<c>true</c> if the texture is newly-bound. <c>false</c> if the texture was already bound.`

File: osu.Framework/Graphics/OpenGL/Textures/TextureGLAtlas.cs [hunk 2]
```diff
@@ -15,48 +16,23 @@ internal class TextureGLAtlas : TextureGLSingle
...
-        /// <summary>
-        /// The total amount of times this <see cref="TextureGLAtlas"/> was bound.
-        /// </summary>
-        public ulong BindCount { get; private set; }
+        public static event Action<TextureGLAtlas> TextureCreated;
 
         public TextureGLAtlas(int width, int height, bool manualMipmaps, All filteringMode = All.Linear)
             : base(width, height, manualMipmaps, filteringMode)
         {
             lock (all_atlases)
```
Comment (smoogipoo) [inline osu.Framework/Graphics/OpenGL/Textures/TextureGLAtlas.cs]: can remove this lock now

File: osu.Framework/Graphics/OpenGL/Textures/TextureGLAtlas.cs [hunk 3]
```diff
@@ -74,8 +50,6 @@ protected override void Dispose(bool isDisposing)
 
             lock (all_atlases)
```
Comment (smoogipoo) [inline osu.Framework/Graphics/OpenGL/Textures/TextureGLAtlas.cs]: And this. Maybe just remove this method altogether?

Comment (peppy) [reply osu.Framework/Graphics/OpenGL/Textures/TextureGLAtlas.cs]: Leaving method for safety (we don't want to expose the raw list).

**Pull request merged**